pkg/handlers/send_odp_event.go

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+ * Copyright 2023, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package handlers //
+package handlers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/optimizely/go-sdk/pkg/odp/event"
+
+	"github.com/optimizely/agent/pkg/middleware"
+	"github.com/optimizely/agent/pkg/optimizely"
+)
+
+// SendOdpEvent sends event to ODP platform
+func SendOdpEvent(w http.ResponseWriter, r *http.Request) {
+	optlyClient, err := middleware.GetOptlyClient(r)
+	if err != nil {
+		RenderError(err, http.StatusInternalServerError, w, r)
+		return
+	}
+
+	body, err := getResponseBody(r)
+	if err != nil {
+		RenderError(err, http.StatusBadRequest, w, r)
+		return
+	}
+
+	success := optlyClient.SendOdpEvent(body.Action, body.Type, body.Identifiers, body.Data)
+	if err != nil {
+		RenderError(err, http.StatusInternalServerError, w, r)
+		return
+	}
+
+	returnResult := optimizely.SendOdpEvent{
```

Comment:
I figured out this is for response formatting. The same name `SendOdpEvent` is confusing with the api call of the same name.

Reply:
Yeah. Renaming to SendOdpEventResponseModel